pkg/ebpf/c/tracee.bpf.c

```diff
@@ -5980,6 +5996,18 @@ CGROUP_SKB_HANDLE_FUNCTION(proto_tcp)
         neteventctx->md.header_size += doff;
     }
 
+    // Check if TCP flow started or ended and submit appropriate event if needed.
+
+    bool flow_sub = should_submit_net_event(neteventctx, SUB_NET_PACKET_TCP_FLOW);
+    bool is_syn_ack = nethdrs->protohdrs.tcphdr.syn && nethdrs->protohdrs.tcphdr.ack;
+    // bool is_fin = nethdrs->protohdrs.tcphdr.fin;
+
+    if (flow_sub && is_syn_ack) {
+        // neteventctx->eventctx.retval |= is_syn_ack ? flow_tcp_begin : flow_tcp_end;
+        neteventctx->eventctx.retval |= flow_tcp_begin;
```

Comment:
As my top review comment says, we're passing this as a flag anyway, so this may not merit a separate base event.

Reply:
Sure, but then I need a filter/config option (from a map) to know if I should submit something that will derive into the flow event.

1) I dont want to submit TCP packets to userland unless they have SYN+ACK or FIN (unless user is also tracing net_packet_XXX). 

2) I need a base event (or config value, or something) to check if I need to submit the packet containing the flags.

What is your suggestion ?

Reply:
I think you could reuse the base tcp packet event and add a derivation function under it.
Recall that we can have multiple derivations for one event, and that we can have separate enable conditions for each derivation function. So you could leave the `SUB_NET_PACKET_TCP_FLOW` as a fake constant for conversion to the actual derived event (which would work internally in `should_submit`), but not define a separate base event. BTW this may merit a separate should_submit function `should_submit_flow`, which can use an enum `flow_type` which would be internal to the eBPF part only, and is only used to convert into event ids (unlike `packet_type` which corresponds one to one with an actual base packet event).

Reply:
I think I prefer something else. To have a generic base event for all the flow events (TCP for now, but we'll have UDP, ICMP, etc) and flag which flow we're deriving through the flags (add flag types for each flow). There will be other flows (DCCP, SCTP, who knows). We could even have vanilla IP flow (first time IP01 talks to IP02, for example). Similar to how netfilter sees flows.

I feel more comfortable doing that I think. Lets see, either way, if the code is working like it should I dont see why holding the merge (we can decide underlyings later). Base events can always be refactored as long as final events dont change.

Reply:
Hum, Im exploring the idea after the refactor I did, lets see, I see your point, just want to make sure it will fit future flow types, etc (maybe a common base event "FlowBase", or just use each protocol Base event like you suggest.. will get back soon on this.